sklearn/metrics/cluster/unsupervised.py

```diff
@@ -342,3 +342,79 @@ def davies_bouldin_score(X, labels):
     score = (intra_dists[:, None] + intra_dists) / centroid_distances
     score[score == np.inf] = np.nan
     return np.mean(np.nanmax(score, axis=1))
+
+
+def score_function(X, labels):
+    """Computes Score Function.
+
+    Parameters
+    ----------
+    X : array-like, shape (``n_samples``, ``n_features``)
+        List of ``n_features``-dimensional data points. Each row corresponds
+        to a single data point.
+
+    labels : array-like, shape (``n_samples``,)
+        Predicted labels for each sample.
+
+    Returns
+    -------
+    sf: float
+        The resulting Score Function.
+
+    References
+    ----------
+    .. [1] Saitta S., Raphael B., Smith I.F.C. (2007).
+        `"A Bounded Index for Cluster Validity"
+        <https://dl.acm.org/citation.cfm?id=1420344>`__.
+        Perner P. (eds) Machine Learning and Data Mining in Pattern Recognition.
+        Lecture Notes in Computer Science, vol 4571.
+    """
+    # Pre-processing and validation of input data and labels
+    X, labels = check_X_y(X, labels)
+    le = LabelEncoder()
+    labels = le.fit_transform(labels)
+
+    n_samples, _ = X.shape
+    n_labels = len(le.classes_)
+
+    check_number_of_labels(n_labels, n_samples)
+
+    # Mean of the entire data. Same as z_tot in the original paper.
+    data_mean = np.mean(X, axis=0)
+
+    # cluster_size : is a 1D-array of length "n_labels" (i.e. number of unique labels or number of unique clusters)
+    # It stores the number of data points in each cluster.
+    cluster_size = np.zeros(n_labels)
+    # intra_dists: It stores the mean euclidean distance between all points belonging to a cluster and their cluster
+    # centroid.
+    # Since there are "n_labels" unique clusters, the variable is a 1D-array of length "n_labels".
+    intra_dists = np.zeros(n_labels)
+    # intra_centroid_dists: It stores the product of euclidean distance between a cluster centroid and the centroid of
```

Comment:
Please limit all lines to a maximum of 79 characters.

Reply:
Thank you for your comments. I will make the necessary changes and get back.